### ISSUE TITLE
watchexec stops executing commands after the second time

### ISSUE THREAD

**Issue by JohnFinn** [watchexec/watchexec#878]
Watchexec stops working after second command execution only for files in `.config`. In `/tmp/` I don't have such bug

https://github.com/user-attachments/assets/f0eda43d-0bfe-494a-8505-f2bbeb09d3a5

- Watchexec's version
```
watchexec 2.1.2 (1980-01-01) +pid1
build-date: 1980-01-01
release: 2.1.2
features: default,pid1
```
- The OS you're using
`Linux nixos 6.6.52 #1-NixOS SMP PREEMPT_DYNAMIC Wed Sep 18 17:24:10 UTC 2024 x86_64 GNU/Linux`
- A log with `-vvv --log-file` [a.log](https://github.com/user-attachments/files/17782264/a.log)
- A sample command that you've run that has the issue
`watchexec --watch ~/.config/foo -vvv --log-file=/tmp/a.log -- cat foo`

**Comment by sakethgetla** [comment]
@JohnFinn for some reason it does not work when watching files but it does work when watching folders.

DOES NOT WORK
~~watchexec -w test/hello.rs "rustc ./test/hello.rs"~~

DOES WORK
` watchexec -w test "rustc ./test/hello.rs"`

**Comment by passcod** [comment]
Oh, you know, I'd completely missed that. **[This is documented](https://github.com/watchexec/watchexec/blob/main/doc/watchexec.1.md#filtering).**

TL;DR: you're watching file. When the file disappears, because your editor implements saving as "create another file, and move it on top of the original, deleting that original", you're no longer watching a file.
